File: leetcode/AIcoding/2.Gemini-2.0/javascript_solutions/3240-minimum-number-of-flips-to-make-binary-grid-palindromic-ii.js
var minFlips = function(grid) {
    const m = grid.length;
    const n = grid[0].length;
    const totalOnes = m * n / 2;
    const target = (1 << totalOnes) - 1;

    const start = 0;
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n / 2; j++) {
            if (grid[i][j] !== grid[i][n - 1 - j]) {
                if (grid[i][j] === 1) {
                    return -1;
                }
            }
        }
    }

    let initialState = 0;
    let count = 0;
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n / 2; j++) {
            if (grid[i][j] === 1) {
                initialState |= (1 << count);
            }
            count++;
        }
    }

    const queue = [[initialState, 0]];
    const visited = new Set();
    visited.add(initialState);

    while (queue.length > 0) {
        const [currState, flips] = queue.shift();

        if (currState === target) {
            return flips;
        }

        for (let i = 0; i < totalOnes; i++) {
            const nextState = currState ^ (1 << i);
            if (!visited.has(nextState)) {
                visited.add(nextState);
                queue.push([nextState, flips + 1]);
            }
        }
    }

    return -1;
};

File: leetcode/AIcoding/2.Gemini-2.0/c_solutions/3240-minimum-number-of-flips-to-make-binary-grid-palindromic-ii.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int minFlips(int** grid, int gridSize, int* gridColSize) {
    int n = gridSize;
    int m = gridColSize[0];

    int initialState = 0;
    for (int i = 0; i < n / 2; i++) {
        for (int j = 0; j < m; j++) {
            if (grid[i][j] != grid[n - 1 - i][j]) {
                initialState |= (1 << (i * m + j));
            }
        }
    }

    int dp[1 << (n * m / 2)];
    for (int i = 0; i < (1 << (n * m / 2)); i++) {
        dp[i] = -1;
    }

    dp[initialState] = 0;

    int queue[1 << (n * m / 2)];
    int head = 0, tail = 0;
    queue[tail++] = initialState;

    while (head < tail) {
        int currState = queue[head++];

        for (int i = 0; i < n / 2; i++) {
            for (int j = 0; j < m; j++) {
                int nextState = currState ^ (1 << (i * m + j));

                if (dp[nextState] == -1) {
                    dp[nextState] = dp[currState] + 1;
                    queue[tail++] = nextState;
                }
            }
        }
    }

    return (dp[0] == -1) ? -1 : dp[0];
}